go/exoservice/exoservice.go

```diff
@@ -93,9 +95,11 @@ func (e *ExoService) ListenForMessages(messageHandlers MessageHandlerMapping) er
 			return nil
 		}
 		go e.receiveMessage(message)
+		e.replyChannelsMutex.RLock()
 		if replyChannel, ok := e.replyChannels[message.ActivityID]; ok {
 			replyChannel <- &message
 		}
+		e.replyChannelsMutex.RUnlock()
```

Comment:
Here you also send the message inside the read lock. Is that intentional? Or is the read lock only there to get the reply channel instance?